### PR TITLE
Added input_netif to the stored data

### PR DESCRIPTION
**Pull request by LaborEtArs**
- Added a 'setMulticastInterface' method for netifs.
- The input_netif is also only available, while 'inside' of the recv-callback. Added this to the temp-saved data.
- Found problems with omitted UDP messages, therefore disabled the the omitting of messages.
- Made '_pcb' accessible

### CODE REVIEW

File: libraries/ESP8266WiFi/src/include/UdpContext.h [hunk 1]
```diff
@@ -441,24 +462,13 @@ class UdpContext
...
-            int count = 0;
-            for (p = _rx_buf; p && ++count < rxBufMaxDepth*2; p = p->next);
-            if (p)
-            {
-                // pbuf chain too deep, dropping
-                pbuf_free(pb);
-                DEBUGV(":udr\r\n");
-                return;
-            }
-        }
```
Comment (d-a-v) [inline libraries/ESP8266WiFi/src/include/UdpContext.h]: did you try to increase `rxBufMaxDepth` instead of removing that block ?

Comment (LaborEtArs) [reply libraries/ESP8266WiFi/src/include/UdpContext.h]: No, might also help, but what would be the right value?
What is the reason to omit UDP messages? Memory usage?

Comment (LaborEtArs) [reply libraries/ESP8266WiFi/src/include/UdpContext.h]: The 'submodules' problem should exist here too, right?

Comment (LaborEtArs) [reply libraries/ESP8266WiFi/src/include/UdpContext.h]: Redid the same process as with the WiFiServer change

Comment (d-a-v) [reply libraries/ESP8266WiFi/src/include/UdpContext.h]: Yes, memory usage

**Comment by LaborEtArs** [comment]
Reinserted the code that omits UDP messages

File: libraries/ESP8266WiFi/src/include/UdpContext.h [hunk 2]
```diff
@@ -167,6 +167,26 @@ class UdpContext
 
 #endif // !LWIP_IPV6
 
+    /*
+     * Add a netif (by its index) as the multicast interface
+     */
+    void setMulticastInterface(netif* p_pNetIf)
+    {
+#if LWIP_VERSION_MAJOR == 1
+        udp_set_multicast_netif_addr(_pcb, (p_pNetIf ? ip_2_ip4(p_pNetIf->ip_addr) : ip_addr_any));
```
Comment (d-a-v) [inline libraries/ESP8266WiFi/src/include/UdpContext.h]: `ip_2_ip4()` is useless with lwIP-1.4
https://github.com/esp8266/Arduino/blob/00440cd84a4ed9467de6305cb768875e09c12303/cores/esp8266/IPAddress.h#L33

File: libraries/ESP8266WiFi/src/include/UdpContext.h [hunk 3]
```diff
@@ -265,7 +290,6 @@ class UdpContext
             // ref'ing it to prevent release from the below pbuf_free(deleteme)
             pbuf_ref(_rx_buf);
         }
-        // remove the already-consumed head of the chain
```
Comment (d-a-v) [inline libraries/ESP8266WiFi/src/include/UdpContext.h]: Perhaps it has to be replaced by
"remove parts of the pbuf chain that are not anymore "ref"ed

**Comment by d-a-v** [comment]
There are unwanted submodule updates in this PR
- libraries/LittleFS/lib/littlefs 
- libraries/SoftwareSerial
- tools/sdk/ssl/bearssl 

(see https://github.com/esp8266/Arduino/pull/7003#issuecomment-572148633)

**Comment by LaborEtArs** [comment]
Can't remove these submodules. Every time I do checkout a new branch, these (and other) submodule are added automagically by git. Closing the PR now.

**Pull request closed without merging**